src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/StripEmbeddedLibraries.cs

```diff
@@ -4,9 +4,192 @@
 using System;
 using System.Linq;
 using Xamarin.Android.Tasks;
+using System.Collections.Generic;
+using Mono.Cecil.Cil;
+using System.Text.RegularExpressions;
 
 namespace MonoDroid.Tuner
 {
+#if !NET5_LINKER
+	public class RemoveResourceDesigner : BaseStep
+	{
+		TypeDefinition mainDesigner = null;
+		AssemblyDefinition mainAssembly = null;
+		Dictionary<string, int> designerConstants;
+		Regex opCodeRegex = new Regex (@"([\w]+): ([\w]+) ([\w.]+) ([\w:./]+)");
+
+		protected override void Process ()
+		{
+			// resolve the MainAssembly Resource designer TypeDefinition
+			foreach(var asm in Context.GetAssemblies ()) {
```

Comment:
.net 5/6 linker api does not have `GetAssemblies`. 
At this point we need to find the "Main" Resource class for the application (i.e IsApplication = true). We need to do this before we process any other assemblies as this class has the final resource id's we need to replace in the references.

Reply:
We can workaround the lack of `GetAssemblies ()` the same way as @rolfbjarne does here: https://github.com/xamarin/xamarin-macios/blob/main/tools/dotnet-linker/Steps/CollectAssembliesStep.cs
Context: https://github.com/mono/linker/issues/1455

Reply:
We had a `AndroidLinkContext` class already, so I assume I will need to add an Assemblies list to that. Then add a new step which will populate it. Then just cast the `_context` as a `AndroidLinkContext` in the ResourceDesignerStep.

Reply:
So AndroidLinkContext is not available in .net5/6. That CollectAssembliesStep in macOS/ios has allot of dependencies. Is there a simpler way other than copying all of that stuff over?

Reply:
I think this was resolved last week in the meeting, right?